File: lib_adapter/src/main/java/com/yyxnb/adapter/ItemDelegate.java
package com.yyxnb.adapter;

public interface ItemDelegate<T> {
    /**
     * 布局文件
     */
    int layoutId();

    /**
     * 类型区分
     */
    boolean isThisType(T item, int position);

    /**
     * 数据处理
     */
    void convert(BaseViewHolder holder, T t, int position);
}

File: view/src/main/java/com/yyxnb/view/rv/ItemDelegate.kt
package com.yyxnb.view.rv

interface ItemDelegate<T> {

    /**
     * 布局文件
     */
    val layoutId: Int

    /**
     * 类型区分
     */
    fun isThisType(item: T, position: Int): Boolean

    /**
     * 数据处理
     */
    fun bind(holder: ViewHolder, t: T, position: Int)

}